doc/userguide/rules/intro.rst

```diff
@@ -69,12 +69,18 @@ Protocol
     alert :example-rule-emphasis:`http` $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP GET Request Containing Rule in URI"; flow:established,to_server; http.method; content:"GET"; http.uri; content:"rule"; fast_pattern; classtype:bad-unknown; sid:123; rev:1;)
 
 This keyword in a signature tells Suricata which protocol it
-concerns. You can choose between four basic protocols:
+concerns. You can choose between several basic protocols:
 
 * tcp (for tcp-traffic)
 * udp
-* icmp
-* ip (ip stands for 'all' or 'any')
+* icmpv4
+* icmpv6
+* icmp (both icmpv4 and icmpv6)
+* sctp
+* gre
+* ip4 or ipv4 ('all' or 'any' IPv4 traffic)
```

Comment:
I know this was in the existing code, but `'all' or 'any'` is a bit misleading as it makes them look like literals that can be used as an alias.  This should probably be:
```
* ip4 or ipv4
* ip6 or ipv6
* ip (both ipv4 and ipv6)
```
The first 2 lines probably don't need the extra annotation as they are as clear as `udp` or `icmp`.